src/log_unstable.rs

```diff
@@ -85,29 +85,19 @@ impl Unstable {
         }
     }
 
-    /// Moves the stable offset up to the index. Provided that the index
-    /// is in the same election term.
-    pub fn stable_to(&mut self, idx: u64, term: u64) {
-        let t = self.maybe_term(idx);
-        if t.is_none() {
-            return;
-        }
-
-        if t.unwrap() == term && idx >= self.offset {
-            let start = idx + 1 - self.offset;
-            self.entries.drain(..start as usize);
-            self.offset = idx + 1;
+    /// Returns the unstable entries and moves the stable offset up to the
+    /// last index, if there is any.
+    pub fn stable_entries(&mut self) -> Vec<Entry> {
```

Comment:
Supposing entries in storage are [(3, 6), (3, 11)], and entries in unstable are [(3, 12), (3, 15)]. After creating a ready, before ready is written to storage, unstable entries become ((3, 16), ∞). At that time a new leader sends conflict entries [(4, 13), (4, 15)], the storage will be queried the term at 13, which doesn't exist yet and then cause panic.